src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -194,9 +195,11 @@ stop_metadata_sync_to_node(PG_FUNCTION_ARGS)
 	CheckCitusVersion(ERROR);
 	EnsureCoordinator();
 	EnsureSuperUser();
+	PreventInTransactionBlock(true, "stop_metadata_sync_to_node");
```

Comment:
currently calling `stop_metadata_sync_to_node` for the coordinator gets into a self deadlock.

```SQL
 SELECT stop_metadata_sync_to_node(nodename,nodeport) FROM pg_dist_node;
DEBUG:  StartTransaction(1) name: unnamed; blockState: DEFAULT; state: INPROGRESS, xid/subid/cid: 0/1/0
NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 0, '2000-01-01 02:00:00+02');
DETAIL:  on server onderkalaci@localhost:5432 connectionId: 27
NOTICE:  issuing SELECT worker_drop_distributed_table(logicalrelid::regclass::text) FROM pg_dist_partition
DETAIL:  on server onderkalaci@localhost:5432 connectionId: 27

```

Besides, we should not try to clear metadata of the coordinator ever :) 

So, use the pattern in `start_metadata_sync_to_node` to skip the coordinator.

```C
if (NodeIsCoordinator(workerNode))
{
..
return/continue whatever needed ;;;
}
```